src/rgw/store/dbstore/common/dbstore.h

```diff
@@ -923,7 +924,7 @@ class ListUserBucketsOp: virtual public DBOp {
                           SwiftVersioning, SwiftVerLocation, \
                           MdsearchConfig, NewBucketInstanceID, ObjectLock, \
                           SyncPolicyInfoGroups, BucketAttrs, BucketVersion, BucketVersionTag, Mtime \
-                          FROM '{}' WHERE OwnerID = {} AND BucketName > {} ORDER BY BucketName ASC LIMIT {}";
+                          FROM '{}' WHERE OwnerID LIKE {} AND BucketName > {} ORDER BY BucketName ASC LIMIT {}";
```

Comment:
1) why are you switching to like?  I would expect this to be less efficient, and we shouldn't need to be doing pattern matching?
2) I didn't notice the use of LIMIT;  not strictly portable, but oh well

Reply:
I switched to "LIKE" to be able to use wildcard '%' and fetch buckets of all the users in the 'ListBuckets' query by GC. But alternatively I can add & use another query for that. Will address it.

All the rgw list operations have a max_limit. I felt its optimal to use it in sqlite query itself than fetching all the entries & then imposing limit in the sal layer all the time.

Reply:
ack, thanks soumya